spec/models/conference_spec.rb

```diff
@@ -1639,4 +1639,26 @@
       expect(free_ticket.price_cents).to eq(0)
     end
   end
+
+  describe 'after_update' do
+    let(:conference) { create(:conference) }
+    let(:first_scheduled_event) { create(:event_scheduled, program: conference.program, hour: conference.start_date + conference.start_hour.hours) }
+    let(:second_scheduled_event) { create(:event_scheduled, program: conference.program, hour: conference.start_date + conference.end_hour.hours + 1.hour) }
+
+    it 'delete event schedules that are not in hour ranges, when conference start hour is updated' do
+      first_scheduled_event
+      conference.start_hour = conference.start_hour + 1.hour
+      conference.save
+      conference.run_callbacks :update
```

Comment:
I was actually following the pattern given here:
https://github.com/siddhantbajaj/osem/blob/007804f3ff4d43dd4ac2137d254cc6e1112839c2/spec/models/conference_spec.rb#L1637
But yes works fine without it as well.